vdom/src/com/vdom/core/CardImplIntrigue.java

```diff
@@ -121,6 +143,42 @@ private void copperSmith(MoveContext context) {
         context.coppersmithsPlayed++;
     }
     
+    private void courtier(Game game, MoveContext context, Player currentPlayer) {
+        if (currentPlayer.getHand().size() == 0) return;
+
+        Card card = currentPlayer.controlPlayer.courtier_cardToReveal(context);
+        if (card == null) {
+            Util.playerError(currentPlayer, "Courtier errror, reveal random card.");
+            card = Util.randomCard(currentPlayer.hand);
+        }
+
+        int numTypes = card.getNumberOfTypes(currentPlayer);
+        if (numTypes > 0) {
+            CourtierOption[] options = currentPlayer.controlPlayer.courtier_chooseOptions(context, numTypes);
```

Comment:
You'll need to pass in a list or Array of CourtierOption here so the player knows what options are valid.